src/styles/main.scss

```diff
@@ -1,3 +1,73 @@
 body {
   margin: 0;
 }
+
+$grey: #eee;
+$square-side: 100px;
+$calendar-gap: 1px;
+$calendar-day-border: 1px;
+$calendar-day-space: $square-side + $calendar-gap;
+$calendar-padding: 10px;
+$calendar-width: ($square-side + $calendar-gap) * 7;
+
+.calendar {
+  width: $calendar-width;
+  margin: 0 auto;
+  display: flex;
+  flex-wrap: wrap;
+  gap: $calendar-gap;
+  padding: $calendar-padding;
+  padding-top: $calendar-padding * 3;
+
+  $days: (
+    0 "mon",
+    1 "tue",
+    2 "wed",
+    3 "thu",
+    4 "fri",
+    5 "sat",
+    6 "sun"
+  );
+
+  @each $number, $day in $days {
+    &--start-day-#{$day} :first-child {
+      margin-left: $number * $calendar-day-space;
+    }
+  }
+
+  @for $day from 28 through 31 {
+    &--month-length--#{$day} {
+      .calendar__day:nth-child(n + #{$day + 1}) {
+        display: none;
+      }
+    }
+  }
+
+  &__day {
+    box-sizing: border-box;
+    background-color: $grey;
+    width: $square-side;
+    height: $square-side;
+    border: $calendar-day-border solid black;
+    display: flex;
+    justify-content: center;
+    align-items: center;
+    transition: 0.5s;
```

Comment:
Make sure to list all styles that you apply transition to - don't use `all` either.

Reply:
> Make sure to list all styles that you apply transition to - don't use `all` either.

Ready!